src/sympc/protocol/falcon/falcon.py

```diff
@@ -356,16 +387,66 @@ def multiplication_protocol(
             x (ReplicatedSharedTensor): Secret
             y (ReplicatedSharedTensor): Another secret
             op_str (str): Operator string.
-            kwargs(Dict[Any, Any]): Keywords arguments for the operator.
+            kwargs (Dict[Any, Any]): Keywords arguments for the operator.
 
         Returns:
             shares (ReplicatedSharedTensor): results in terms of ReplicatedSharedTensor.
         """
-        op = getattr(operator, op_str)
-
-        z_value = (
-            op(x.shares[0], y.shares[0], **kwargs)
-            + op(x.shares[1], y.shares[0], **kwargs)
-            + op(x.shares[0], y.shares[1], **kwargs)
+        op = ReplicatedSharedTensor.get_op(x.ring_size, op_str)
+
+        z_value = shares_sum(
+            [
+                op(x.shares[0], y.shares[0], **kwargs),
+                op(x.shares[1], y.shares[0], **kwargs),
+                op(x.shares[0], y.shares[1], **kwargs),
+            ],
+            x.ring_size,
         )
         return z_value
+
+    @staticmethod
+    def select_shares(x: MPCTensor, y: MPCTensor, b: MPCTensor) -> MPCTensor:
+        """Returns either x or y based on bit b.
+
+        Args:
+            x (MPCTensor): input tensor
+            y (MPCTensor): input tensor
+            b (MPCTensor): input tensor which is shares of a bit used as selector bit.
+
+        Returns:
+            z (MPCTensor):Returns x (if b==0) or y (if b==1).
+
+        Raises:
+            ValueError: If the selector bit tensor is not of ring size "2".
+        """
+        ring_size = int(b.share_ptrs[0].get_ring_size().get_copy())
+        shape = b.shape
+        if ring_size != 2:
+            raise ValueError(
+                f"Invalid {ring_size} for bit_injection,must be of ring size 2"
+            )
+        if shape is None:
+            raise ValueError("The selector bit tensor must have a valid shape.")
+        session = x.session
+        c_ptrs: List[ReplicatedSharedTensor] = []
```

Comment:
MAYBE: Add a TODO such that we generate those using the CryptoProvider?

Reply:
Sure George :+1: